### PR TITLE
Add Rubocop

### PR DESCRIPTION
**Pull request by mamhoff**
This adds Rubocop to the project and applies its recommendation to all code files. Yay!

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+CollectionMethods:
+  Enabled: false
+
+# Sometimes I believe this reads better
+# This also causes spacing issues on multi-line fixes
+Style/BracesAroundHashParameters:
+  Enabled: false
+
+# We use class vars and will have to continue doing so for compatability
```
Comment (tvdeyen) [inline .rubocop.yml]: Is this true?

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+Style/ClassVars:
+  Enabled: false
+
+# We need these names for backwards compatability
```
Comment (tvdeyen) [inline .rubocop.yml]: Don’t think this applies to this gem

File: .rubocop.yml [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+  Enabled: false
+
+Naming/AccessorMethodName:
+  Enabled: false
+
+# This has been used for customization
+Style/MutableConstant:
+  Enabled: false
+
+# `something.count > 0` is often used in Solidus admin and IMO better to read then `something.count.positive?`
```
Comment (tvdeyen) [inline .rubocop.yml]: Don’t think this applies to this gem

File: .circleci/config.yml [hunk 4]
```diff
@@ -4,7 +4,30 @@
 #
 version: 2
 jobs:
-  build:
+  RuboCop:
+    working_directory: ~/cspree
```
Comment (tvdeyen) [inline .circleci/config.yml]: 😏

File: .rubocop.yml [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+  Enabled: false
+
+# We need these names for backwards compatability
+Naming/PredicateName:
+  Enabled: false
+
+Naming/AccessorMethodName:
+  Enabled: false
+
+# This has been used for customization
```
Comment (tvdeyen) [inline .rubocop.yml]: Don’t think this applies to this gem

File: .rubocop.yml [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+#     expect {
+#       something
+#     }.to { happen }
+#
+# syntax in the specs files.
+Lint/AmbiguousBlockAssociation:
+  Exclude:
+    - 'spec/**/*'
+
+# We use eval to add common_spree_dependencies into the Gemfiles of each of our gems
```
Comment (tvdeyen) [inline .rubocop.yml]: Don’t think this applies to this gem

File: .rubocop.yml [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+    - '*/Gemfile'
+
+Naming/VariableNumber:
+  Enabled: false
+
+# Write empty methods as you wish.
+Style/EmptyMethod:
+  Enabled: false
+
+# The Rails Guides are in contradiction with this cop.
```
Comment (tvdeyen) [inline .rubocop.yml]: Don’t think this applies to this gem

File: .rubocop.yml [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+Style/FormatStringToken:
+  Exclude:
+    - 'config/routes.rb'
+
+# From http://relaxed.ruby.style/
```
Comment (tvdeyen) [inline .rubocop.yml]: I would recommend to remove all disabled rules from above and only keep the relaxed styles (maybe update them from current version)

File: spec/friendly_shipping/services/ship_engine/serialize_rate_estimate_request_spec.rb [hunk 9]
```diff
@@ -10,16 +12,16 @@
...
+                           carrier_ids: ["se-123456"],
+                           from_country_code: "US",
+                           from_postal_code: shipment.origin.zip,
+                           to_country_code: "US",
+                           to_postal_code: shipment.destination.zip,
+                           to_city_locality: "Herndon",
+                           to_state_province: "IL",
+                           weight: {value: 0.0625, unit: "pound"},
+                           confirmation: "none",
+                           address_residential_indicator: "no"
```
Comment (tvdeyen) [inline spec/friendly_shipping/services/ship_engine/serialize_rate_estimate_request_spec.rb]: Yikes!

**Comment by mamhoff** [comment]
I added the relaxed ruby style guide instead, and only disabled the block length cop - due to the use of Nokogiri and RSpec, it usually gives false negatives.

File: lib/friendly_shipping/services/ups/shipping_methods.rb [hunk 10]
```diff
@@ -80,11 +85,11 @@ def countries_by_code(code)
...
-        ['OTHER', 'international','UPS Worldwide Express Plus', '54'],
-        ['OTHER', 'international','UPS Worldwide Saver', '65'],
+        ['OTHER', 'domestic', 'UPS Express', '07'],
+        ['OTHER', 'domestic', 'UPS Standard', '11'],
+        ['OTHER', 'international', 'UPS Worldwide Expedited', '08'],
+        ['OTHER', 'international', 'UPS Worldwide Express Plus', '54'],
+        ['OTHER', 'international', 'UPS Worldwide Saver', '65'],
         ['ALL', 'international', 'UPS Worldwide Express Freight', '96'],
         ['ALL', 'international', 'UPS Worldwide Express Freight Midday', '71']
       ].map do |origin_country_code, dom_or_intl, name, code|
```
Comment (tvdeyen) [inline lib/friendly_shipping/services/ups/shipping_methods.rb]: I think in a later PR this list of country mapping could be a frozen constant as well.

File: spec/friendly_shipping/services/ship_engine/parse_rate_estimate_response_spec.rb [hunk 11]
```diff
@@ -21,7 +23,7 @@
     expect(carrier).to eq(carriers.first)
     expect(rate.amounts.keys).to contain_exactly(:confirmation, :insurance, :other, :shipping)
     expect(rate.total_amount).to eq(Money.new(535, 'USD'))
-    expect(rate.delivery_date).to eq(Time.new(2019, 06, 07, 00, 00, 00, "+00:00"))
+    expect(rate.delivery_date).to eq(Time.new(2019, 0o6, 0o7, 0o0, 0o0, 0o0, "+00:00"))
```
Comment (tvdeyen) [inline spec/friendly_shipping/services/ship_engine/parse_rate_estimate_response_spec.rb]: Funny ASCIImojis :)

File: lib/friendly_shipping/services/ups/shipping_methods.rb (hunk 10 above)
Comment (mamhoff) [reply lib/friendly_shipping/services/ups/shipping_methods.rb]: Frozen in this PR

File: spec/friendly_shipping/services/ship_engine/parse_rate_estimate_response_spec.rb (hunk 11 above)
Comment (mamhoff) [reply spec/friendly_shipping/services/ship_engine/parse_rate_estimate_response_spec.rb]: Fixed :)

**Pull request merged**